### DISCUSSION TITLE
Redirect from sign in pages when already signed in

### DISCUSSION THREAD

**Discussion by StevenClontz** [mikker/passwordless#233] [General]
Is there a slick way I can redirect from `/users/sign_in` to `/` (or wherever) when the user is already authenticated?

**Comment by mikker** [comment]
You could do with a `before_action` filter i `ApplicationController` or maybe a route constraint or you could write your own `SessionsController#new` controller/action.

**Comment by StevenClontz** [comment]
Thanks! I ended up going this route:

```ruby
# config/routes.rb
passwordless_for :users, controller: "sessions"
```

```ruby
# app/controllers/sessions_controller.rb

class SessionsController < Passwordless::SessionsController
  before_action :require_unauth!, only: %i[ new show ]

  private

    def require_unauth!
      unless @current_user.nil?
        redirect_to "/", notice: "You are already signed in."
      end
    end
end
```

**Comment by mikker** [comment]
`unauth` 😅 Looks like a good solution! Thank you for contributing it back to the project

**Reply by StevenClontz** [reply]
No problem! This gem has done exactly what I hoped it would for my own project, so thank you.
